File: java/com/countrygamer/angel/common/block/WABlocks.java
package com.countrygamer.angel.common.block;

import net.minecraft.block.Block;
import net.minecraft.block.material.Material;
import net.minecraft.tileentity.TileEntity;

import com.countrygamer.angel.common.WeepingAngels;
import com.countrygamer.angel.common.tile.TileEntityPlinth;
import com.countrygamer.countrygamercore.Base.Plugin.registry.PluginBlockRegistry;
import com.countrygamer.countrygamercore.common.Core;

import cpw.mods.fml.common.registry.GameRegistry;

public class WABlocks implements PluginBlockRegistry {
	
	public static Block plinthBlock;
	public static Block blockWeepingAngelSpawn;
	
	@Override
	public void registryTileEntities() {
		TileEntity.addMapping(TileEntityPlinth.class, WeepingAngels.PLUGIN_ID + "_Plinth");
		
	}
	
	@Override
	public void registerBlocks() {
		plinthBlock = new BlockPlinth(TileEntityPlinth.class, Material.rock);
		plinthBlock.setHardness(2.0F).setResistance(10F);
		plinthBlock.setStepSound(Block.soundTypeStone);
		plinthBlock.setBlockName("Plinth");
		GameRegistry.registerBlock(plinthBlock, "Plinth");
		
		blockWeepingAngelSpawn = new BlockWeepingAngelSpawn().setHardness(0.5F);
		blockWeepingAngelSpawn.setBlockName("weepingangelspawn");
		Core.addBlockToTab(WABlocks.blockWeepingAngelSpawn);
		GameRegistry.registerBlock(blockWeepingAngelSpawn,
				"Weeping Angel Spawn Block");
	}
	
	@Override
	public void registerBlockCraftingRecipes() {
		
	}
	
	@Override
	public void registerBlockSmeltingRecipes() {
		
	}
	
	@Override
	public void registerOtherBlockRecipes() {
		
	}
	
}

File: src/main/scala/com/countrygamer/weepingangels/common/init/WABlocks.scala
package com.countrygamer.weepingangels.common.init

import com.countrygamer.cgo.library.common.Origin
import com.countrygamer.cgo.library.common.register.BlockRegister
import com.countrygamer.weepingangels.common.WeepingAngels
import com.countrygamer.weepingangels.common.block.BlockStatue
import com.countrygamer.weepingangels.common.tile.TileEntityStatue
import cpw.mods.fml.common.registry.GameRegistry
import net.minecraft.block.Block
import net.minecraft.block.material.Material

/**
 *
 *
 * @author CountryGamer
 */
object WABlocks extends BlockRegister {

	var statue: Block = null

	override def registerTileEntities: Unit = {

		GameRegistry
				.registerTileEntity(classOf[TileEntityStatue], WeepingAngels.pluginID + "_Statue")

	}

	override def register(): Unit = {

		WABlocks.statue = new BlockStatue(Material.rock, WeepingAngels.pluginID, "Plinth",
			classOf[TileEntityStatue])
		Origin.addBlockToTab(WABlocks.statue)

	}

	override def registerCrafting: Unit = {

	}

	override def registerSmelting: Unit = {

	}

	override def registerOther: Unit = {

	}

}
